validator_client/src/duties_service.rs

```diff
@@ -818,24 +818,32 @@ async fn poll_beacon_attesters_for_epoch<T: SlotClock + 'static, E: EthSpec>(
         return Ok(());
     }
 
-    // Filter out validators which have already been requested.
-    let initial_duties = &response.data;
+    // Make a request for all `validators_to_update`, even if we already requested duties for them
+    // in the initial query. There was previously a bug here where we assumed the duties from the
+    // initial query were "new" and needed to be inserted into the map, which overrode information
+    // in the `subscription_slots` and caused expired subscriptions to be sent.
+    // Seeing as the initial batch is so small, the worst-case bandwidth wastage here is minimal,
```

Comment:
Wouldn't `uninitialized_validators.len() == local_pubkeys.len()` right after a restart?
So we would end up making 2 requests with `indices_to_request.len() == uninitialized_validators.len()` in that case.
I think this is okay, but just wanted to confirm that this would be the case.

Reply:
Oh yeah you are right! I got a bit carried away with the 1 validator case

Will try to fix it up so we do reuse the info from the first query, but don't attempt any overrides

Reply:
Pushed that update. I still want to roll the changes from the other PR into this one to handle the startup case. I noticed a bunch of warnings on restart when I deployed this PR.